service/geopmdpy/service.py

```diff
@@ -219,6 +231,14 @@ def _write_mode(self, client_pid):
             # daemon restart
             self._pio.save_control()
 
+    def _watch_client(self, client_pid):
+        return GLib.timeout_add(self._WATCH_INTERVAL_MSEC, self._check_client, client_pid)
+
+    def _check_client(self, client_pid):
+        if client_pid in self._sessions and not psutil.pid_exists(client_pid):
+            self.close_session(client_pid)
```

Comment:
The `return False` causes the timeout to be destroyed, and close_session also removes the timeout source. Is it okay for both to happen?

Reply:
I was worried about this too.  If you call GObj.source_remove() and the handle you pass is already removed, a warning is printed.  If, however, you call GObj.source_remove(), and then the callback returns False, there is no warning or error emitted.